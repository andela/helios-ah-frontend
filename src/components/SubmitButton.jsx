import React from 'react';
import PropTypes from 'prop-types';

const SubmitButton = ({
  onClick,
  isRequestSent,
  value,
  id,
  className,
  columnAttribute
}) => (
  <div className={columnAttribute}>
    <div className="form-group text-center">
      <button
        type="submit"
        className={className}
        id={id}
        onClick={onClick}
        disabled={isRequestSent}
      >
        {value}
      </button>
    </div>
  </div>
);

SubmitButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  isRequestSent: PropTypes.bool,
  value: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  columnAttribute: PropTypes.string,
};

SubmitButton.defaultProps = {
  isRequestSent: false,
  columnAttribute: ''
};

export default SubmitButton;
