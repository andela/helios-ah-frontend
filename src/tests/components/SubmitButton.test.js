import React from 'React';
import { shallow } from 'enzyme';
import SubmitButton from '../../components/SubmitButton';
describe('Render Submit button', () => {
  let wrapper;
  let props;
  beforeAll(() => {
    props = {
      submitDetails: jest.fn(),
      isRequestSent: true,
      buttonValue: 'SIGNUP',
      idAttribute: 'btn-test',
      buttonClass: 'btn',
      columnAttribute: 'signup-col'
    }
    wrapper = shallow(<SubmitButton {...props} />)
  })

  it('it should render submit button', () => {
    expect(wrapper).toMatchSnapshot();
  });
});