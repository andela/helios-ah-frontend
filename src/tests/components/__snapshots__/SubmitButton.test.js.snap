// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Submit button it should render submit button 1`] = `
<div
  className="signup-col"
>
  <div
    className="form-group text-center"
  >
    <button
      className="btn"
      disabled={true}
      id="btn-test"
      onClick={[MockFunction]}
      type="submit"
    >
      SIGNUP
    </button>
  </div>
</div>
`;
